### DISCUSSION TITLE
Comparison / Benchmark between UWS and H2O?

### DISCUSSION THREAD

**Discussion by joshxyzhimself** [uNetworking/uWebSockets#1403] [Q&A] [answered]
Hi all, I have noticed that TypeSense [1] uses H2O [2] as their HTTP server.

According to [3]:

> Typesense comes built-in with a high performance HTTP server [2] that is used by likes of Fastly [4] in their edge servers at scale. So Typesense can be directly exposed to incoming public-facing internet traffic, without the need to place it behind another web server like Nginx / Apache or your backend API.

Has anyone compared / benchmarked UWS and H2O before?

Just curious, thank you very much.

Update: UWS benchmarks are at [5], H2O benchmarks are at [6].

Update: Found an old issue about UWS and H2O at [7].

- [1] https://typesense.org/
- [2] https://github.com/h2o/h2o 
- [3] https://typesense.org/docs/guide/running-in-production.html#production-best-practices
- [4] https://www.fastly.com/
- [5] https://github.com/uNetworking/uWebSockets/tree/master/benchmarks
- [6] https://h2o.examp1e.net/benchmarks.html
- [7] https://github.com/h2o/h2o/issues/1228

**Comment by ghost** [comment, marked as answer]
It's literally in the graph you posted yourself. See 5.

**Reply by joshxyzhimself** [reply]
Thank you. I need to get my eyes checked :laughing:
